### PR TITLE
Fixes #23799: Make rudderc able to compile techniques using a per method policy mode override

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23799

### CODE REVIEW

File: policies/rudderc/tests/cases/general/policy_mode/technique.cf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "46b8025a-0b06-485c-9127-50e4258ee7e6_${report_data.directive_id}" usebundle => call_test_audit_46b8025a_0b06_485c_9127_50e4258ee7e6("Package present", "vim", "46b8025a-0b06-485c-9127-50e4258ee7e6", @{args}, "${class_prefix}", "vim", "", "", "");
+
+    "1eedce7b-3441-4251-bdd6-706fda3ec7a8_${report_data.directive_id}" usebundle => call_test_audit_1eedce7b_3441_4251_bdd6_706fda3ec7a8("Package present", "htop", "1eedce7b-3441-4251-bdd6-706fda3ec7a8", @{args}, "${class_prefix}", "htop", "", "", "");
+
+}
+bundle agent call_test_audit_46b8025a_0b06_485c_9127_50e4258ee7e6(c_name, c_key, report_id, args, class_prefix, name, version, architecture, provider) {
+
+  methods:
+    "46b8025a-0b06-485c-9127-50e4258ee7e6_${report_data.directive_id}" usebundle => _method_reporting_context_v4("${c_name}", "${c_key}", "${report_id}");
+    "46b8025a-0b06-485c-9127-50e4258ee7e6_${report_data.directive_id}" usebundle => push_dry_run_mode(true);
```
Comment (Fdall) [inline policies/rudderc/tests/cases/general/policy_mode/technique.cf:22]: This need to be "true" instead

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Pull request merged**